service/history/handler.go

```diff
@@ -237,6 +254,11 @@ func (h *Handler) IsWorkflowTaskValid(ctx context.Context, request *historyservi
 		return nil, h.convertError(err)
 	}
 
+	startTime := h.timeSource.Now()
+	defer func() {
+		h.healthSignalAggregator.Record("IsWorkflowTaskValid", time.Since(startTime), retError)
+	}()
```

Comment:
Is it possible to have it as an interceptor?

Reply:
We would have to pull from prometheus and we risk it being down. Let me know if I'm misunderstanding the suggestion